sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -244,28 +245,31 @@ class NodeHttpClient implements HttpClient {
   }
 
   private getOrCreateAgent(request: PipelineRequest, isInsecure: boolean): http.Agent {
-    if (!request.disableKeepAlive) {
-      if (isInsecure) {
-        if (!this.httpKeepAliveAgent) {
-          this.httpKeepAliveAgent = new http.Agent({
-            keepAlive: true,
-          });
-        }
+    if (!request.tlsSettings && request.disableKeepAlive) {
+      return isInsecure ? http.globalAgent : https.globalAgent;
+    }
+    const settingsIndex = request.tlsSettings ?? {};
```

Comment:
I think we should pull up the logic for the case of insecure + keepAlive here, instead of combining with the lower codepath, since we know we're going to ignore tlsSettings for any insecure connection - we can simply initialize the httpKeepAliveAgent agent if it doesn't exist and return it